core/amber/src/main/scala/edu/uci/ics/texera/web/service/ExecutionResultService.scala

```diff
@@ -75,38 +62,39 @@ object ExecutionResultService {
     *
     * Produces the WebResultUpdate to send to frontend from a result update from the engine.
     */
-  def convertWebResultUpdate(
+  private def convertWebResultUpdate(
       sink: ProgressiveSinkOpDesc,
       oldTupleCount: Int,
       newTupleCount: Int
   ): WebResultUpdate = {
     val webOutputMode: WebOutputMode = {
-      (sink.getOutputMode, sink.getChartType) match {
-        // visualization sinks use its corresponding mode
-        case (SET_SNAPSHOT, Some(_)) => SetSnapshotMode()
-        case (SET_DELTA, Some(_))    => SetDeltaMode()
-        // Non-visualization sinks use pagination mode
-        case (_, None) => PaginationMode()
+      sink.getOutputMode match {
+        // currently, only table outputs are using these modes
+        case OutputMode.SET_DELTA    => SetDeltaMode()
+        case OutputMode.SET_SNAPSHOT => PaginationMode()
+
+        // currently, only visualizations are using single snapshot mode
+        case OutputMode.SINGLE_SNAPSHOT => SetSnapshotMode()
       }
     }
 
     val storage =
       ResultStorage.getOpResultStorage(sink.getContext.workflowId).get(sink.getUpstreamId.get)
-    val webUpdate = (webOutputMode, sink.getOutputMode) match {
-      case (PaginationMode(), SET_SNAPSHOT) =>
+    val webUpdate = webOutputMode match {
+      case PaginationMode() =>
         val numTuples = storage.getCount
         val maxPageIndex =
-          Math.ceil(numTuples / ExecutionResultService.defaultPageSize.toDouble).toInt
+          Math.ceil(numTuples / defaultPageSize.toDouble).toInt
         WebPaginationUpdate(
           PaginationMode(),
           newTupleCount,
           (1 to maxPageIndex).toList
         )
-      case (SetSnapshotMode(), SET_SNAPSHOT) =>
-        tuplesToWebData(webOutputMode, storage.get().toList, sink.getChartType)
-      case (SetDeltaMode(), SET_DELTA) =>
+      case SetSnapshotMode() =>
+        tuplesToWebData(webOutputMode, storage.get().toList)
+      case SetDeltaMode() =>
         val deltaList = storage.getAfter(oldTupleCount).toList
-        tuplesToWebData(webOutputMode, deltaList, sink.getChartType)
+        tuplesToWebData(webOutputMode, deltaList)
 
       // currently not supported mode combinations
       // (PaginationMode, SET_DELTA) | (DataSnapshotMode, SET_DELTA) | (DataDeltaMode, SET_SNAPSHOT)
```

Comment:
fix this comment

Reply:
removed as it is not accurate any more. we don't do combinations any more.